Saule/Serialization/ResourceSerializer.cs

```diff
@@ -32,11 +33,26 @@ internal class ResourceSerializer
             IncludeContext includeContext,
             FieldsetContext fieldsetContext,
             IPropertyNameConverter propertyNameConverter = null)
+            : this(value, value, type, baseUrl, urlBuilder, paginationContext, includeContext, fieldsetContext, propertyNameConverter)
+        {
+        }
+
+        public ResourceSerializer(
+            object value,
+            object unwrappedValue,
```

Comment:
the name implies here that `unwrappedValue` is somehow derived from `value`. So could we pass just `value` here? or you think the unwrapping is expensive? I think i need some more context to understand this part.

Reply:
Commented this point above too. We need to have both of original value and unwrapped one. as if we have `PagedResult<T>` then we can't apply sort\page\filter operations on top of it, as we need to use actual items that were returned. And `PagedResult<T>` won't be returned in the response as it will be used like meta data to understand total items count and use it to generate `links` section